packages/shared/src/components/post/write/CreatePostButton.tsx

```diff
@@ -36,11 +36,11 @@ export function CreatePostButton({
   const handle = route === '/squads/[handle]' ? (query.handle as string) : '';
   const { squad } = useSquad({ handle });
   const allowedToPost = verifyPermission(squad, SourcePermissions.Post);
-  const hasAccess = squads?.some((item) =>
-    verifyPermission(item, SourcePermissions.Post),
-  );
+  const hasAccess = handle
+    ? squads?.some((item) => verifyPermission(item, SourcePermissions.Post))
+    : true;
```

Comment:
When there is a handle, it indicates the user is at the actual post-writing page, which we need validation. But if it is `undefined` it means we are at any other page in which the header "Post" button must be enabled.